Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1123 @@
+                                else
+                                {
+                                    $apiError = $true;
+                                    Write-Host "Error fetching Front Door configuration..." -ForegroundColor $([Constants]::MessageType.Warning)
+                                    Write-Host $([Constants]::SingleDashLine)
+                                }        
+                            }
+                            if(($null -ne $MinTLSVersion) -and ($apiError -eq $false))
+                            {
+                                if($MinTLSVersion -ne $requiredMinTLSVersion)
+                                {
+                                    $frontDoorEp.DomainName = $domainName
+                                    $frontDoorEp.FrontDoorName = $resourceName
+                                    $frontDoorEp.MinTLSVersion = $MinTLSVersion
+                                    $frontDoorEp.ResourceGroup = $resourceGroupName
+                                    $frontDoorEp.TypeOfCertificate = $TypeOfCertificate
+                                    $frontDoorEp.isMinTLSVersionSetOnCustomDomain = $false
+                                    
+                                    if($null -ne $frontDoorEp)
+                                    {
+                                        $nonCompliantFrontDoorEPs +=  $frontDoorEp | Select-Object @{N='ResourceGroupName';E={$frontDoorEp.ResourceGroup}},
+                                                                                        @{N='ResourceName';E={$frontDoorEp.FrontDoorName}},
+                                                                                        @{N='DomainName';E={$frontDoorEp.DomainName}},
+                                                                                        @{N='CertificateType';E={$frontDoorEp.TypeOfCertificate}},
+                                                                                        @{N='MinimumTlsVersion';E={$frontDoorEp.MinTLSVersion}},
+                                                                                        @{N='isMinTLSVersionSetOnCustomDomain';E={$frontDoorEp.isMinTLSVersionSetOnCustomDomain}}
+                                    }
+                                }
+                                else
+                                {
+                                    $compliantFrontDoorEPs += $item
+                                }
+                            }
+                            else
+                            {
+                                Write-Host "Skipping this Front End Point..." -ForegroundColor $([Constants]::MessageType.Warning)
+                                Write-Host $([Constants]::SingleDashLine)
+                            }
+                        }
+                }
+              }
+              else
+              {
+                if($frontDoorResource.MinimumTlsVersion -ne $requiredMinTLSVersion)
```

Comment:
In the If block we are checking minimum TLS version of endpoint while in else we are checking minimum TLS version at front door level. Is this expected?

Reply:
We are checking this here because if the user has used file path this variable frontDoorResource will be already having minimum tls version so if it is null it will goto if block otherwise we will directly add it to the noncompliantEP variable.